pkg/model/insight.proto

```diff
@@ -56,3 +56,50 @@ enum InsightApplicationCountLabelKey {
   KIND = 0;
   ACTIVE_STATUS = 1;
 }
+
+message InsightDeploymentSubset {
+  int64 created_at = 14 [(validate.rules).int64.gte = 0];
+  int64 updated_at = 15 [(validate.rules).int64.gte = 0];
+}
+
+message InsightDailyDeployment {
+  // `date` must be normalized like xxxx/xx/xx:00:00:00;
+  int64 date = 1 [(validate.rules).int64.gte = 0];
+  int64 created_at = 14 [(validate.rules).int64.gte = 0];
+  int64 updated_at = 15 [(validate.rules).int64.gte = 0];
+
+  // Daily InsightDeploymentSubset sorted by created_at ASC.
+  repeated InsightDeploymentSubset daily_deployments = 4;
+}
+
+// InsightDeploymentChunk represents a chunk that is used by insight
+message InsightDeploymentChunk {
+  InsightChunkDateRange date_range = 1;
+
+  // Daily-deployments sorted by InsightDailyDeployment.date ASC.
+  repeated InsightDailyDeployment deployments = 2;
+}
+
+message InsightDeploymentChunkMetaData {
```

Comment:
Where is this being used?

Reply:
This(and `InsightDeploymentChunk`) will be also stored in filestore.
In this PR, this is only defined and not used.